src/main/java/seedu/address/logic/commands/MarkTaskCommand.java

```diff
@@ -34,15 +34,15 @@ public MarkTaskCommand(Index targetIndex) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Task> lastShownList = model.getFilteredTaskList();
+        List<Task> lastShownList = model.getSortedTaskList();
```

Comment:
Does this work properly for marking a task after searching for it?

Reply:
`model.getSortedTaskList()` represents the final items on the screen(?). Resolved offline.